src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/Fixers/SourceGeneratorAttributeAnalyzerFix.cs

```diff
@@ -0,0 +1,114 @@
+        public const string CSharpVisualBasicEquivalenceKey = "GeneratorFixCSharpVisualBasic";
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(
+            DiagnosticIds.MissingSourceGeneratorAttributeId);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span);
+
+            if (node is null)
+            {
+                return;
+            }
+
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                AddFix(
+                    string.Format(CodeAnalysisDiagnosticsResources.AddGeneratorAttribute_1, LanguageNames.CSharp),
+                    context, document, node, diagnostic, CSharpEquivalenceKey, LanguageNames.CSharp);
+
+                AddFix(
+                    string.Format(CodeAnalysisDiagnosticsResources.AddGeneratorAttribute_1, LanguageNames.VisualBasic),
+                    context, document, node, diagnostic, VisualBasicEquivalenceKey, LanguageNames.VisualBasic);
+
+                AddFix(
+                    string.Format(CodeAnalysisDiagnosticsResources.AddGeneratorAttribute_2, LanguageNames.CSharp, LanguageNames.VisualBasic),
+                    context, document, node, diagnostic, CSharpVisualBasicEquivalenceKey, LanguageNames.CSharp, LanguageNames.VisualBasic);
+            }
+        }
+
+        private static void AddFix(string title, CodeFixContext context, Document document, SyntaxNode node, Diagnostic diagnostic, string equivalenceKey, params string[] languageNames)
+        {
+            var codeAction = CodeAction.Create(
+                title,
+                (cancellationToken) => FixDocumentAsync(document, node, languageNames, cancellationToken),
+                equivalenceKey: equivalenceKey);
+
+            context.RegisterCodeFix(codeAction, diagnostic);
+        }
+
+        public override FixAllProvider? GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        private static async Task<Document> FixDocumentAsync(Document document, SyntaxNode node, string[] languageNames, CancellationToken cancellationToken)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            var generator = editor.Generator;
+
+            SyntaxNode? generatorAttribute;
+
+            if (languageNames.Length == 1 && languageNames[0] == LanguageNames.CSharp)
+            {
+                // CSharp is the only language, which is the default paramterless 
+                // constructor for the Generator attribute
+                generatorAttribute = generator.Attribute(WellKnownTypeNames.MicrosoftCodeAnalysisGeneratorAttribute).WithAddImportsAnnotation();
```

Comment:
Looks like imports annotation isn't working as I was expecting (`TestSimpleClass_FullyQualified_CSharp` should have failed).

I think Roslyn can improve this by assuming the annotation is meant to any descendant nodes, but for now we can generate the type syntax ourselves and pass it to `generator.Attribute` similar to


https://github.com/dotnet/roslyn-analyzers/blob/c3145214555afda60a74da31f9096a788348e4fc/src/Roslyn.Diagnostics.Analyzers/Core/ImportingConstructorShouldBeObsoleteCodeFixProvider.cs#L86-L91

https://github.com/dotnet/roslyn-analyzers/blob/c3145214555afda60a74da31f9096a788348e4fc/src/Roslyn.Diagnostics.Analyzers/Core/ImportingConstructorShouldBeObsoleteCodeFixProvider.cs#L103-L104